### PR TITLE
tests: add aws_instance testcase

### PR DESCRIPTION
**Pull request by syuu1228**
Implement aws_instance testcase, and run it on GitHub Actions.

Also fix two bugs on aws_instance found in the testcase.

### CODE REVIEW

File: tests/test_aws_instance.py [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+    def test__non_root_nvmes_i3en_2xlarge(self):
+        self.httpretty_aws_metadata()
+        with unittest.mock.patch('psutil.disk_partitions', return_value=mock_disk_partitions),\
+            unittest.mock.patch('os.listdir', return_value=mock_listdevdir_i3en_2xlarge),\
+            unittest.mock.patch('lib.scylla_cloud.run', side_effect=mock_multi_run_i3en_2xlarge),\
+            unittest.mock.patch('builtins.open', unittest.mock.MagicMock(side_effect=mock_multi_open_i3en_2xlarge)):
+            ins = aws_instance()
+            assert ins._non_root_nvmes() == {'root': ['/dev/nvme0n1p1'], 'ephemeral': ['nvme2n1', 'nvme1n1'], 'ebs': []}
+
+    def test___populate_disks_i3en_2xlarge(self):
```
Comment (fruch) [inline tests/test_aws_instance.py:309]:
```suggestion
    def test_populate_disks_i3en_2xlarge(self):
```

File: tests/test_aws_instance.py [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+        ins = aws_instance()
+        assert ins.user_data == '{"scylla_yaml": {"cluster_name": "test-cluster"}}'
+
+    def test_no_user_data(self):
+        self.httpretty_aws_metadata()
+        ins = aws_instance()
+        assert not ins.user_data
+
+
+    def test__non_root_nvmes_i3en_2xlarge(self):
```
Comment (fruch) [inline tests/test_aws_instance.py:300]:
```suggestion
    def test_non_root_nvmes_i3en_2xlarge(self):
```

File: tests/test_aws_instance.py [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+    def test__non_root_nvmes_i3en_2xlarge_with_ebs(self):
+        self.httpretty_aws_metadata(with_ebs=True)
+        with unittest.mock.patch('psutil.disk_partitions', return_value=mock_disk_partitions),\
+            unittest.mock.patch('os.listdir', return_value=mock_listdevdir_i3en_2xlarge_with_ebs),\
+            unittest.mock.patch('lib.scylla_cloud.run', side_effect=mock_multi_run_i3en_2xlarge),\
+            unittest.mock.patch('builtins.open', unittest.mock.MagicMock(side_effect=mock_multi_open_i3en_2xlarge_with_ebs)):
+            ins = aws_instance()
+            assert ins._non_root_nvmes() == {'root': ['/dev/nvme0n1p1'], 'ephemeral': ['nvme3n1', 'nvme4n1'], 'ebs': ['nvme2n1', 'nvme1n1']}
+
+    def test___populate_disks_i3en_2xlarge_with_ebs(self):
```
Comment (fruch) [inline tests/test_aws_instance.py:393]:
```suggestion
    def test_populate_disks_i3en_2xlarge_with_ebs(self):
```

File: tests/test_aws_instance.py [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+    def test__non_root_nvmes_i3_2xlarge(self):
+        self.httpretty_aws_metadata()
+        with unittest.mock.patch('psutil.disk_partitions', return_value=mock_disk_partitions),\
+            unittest.mock.patch('os.listdir', return_value=mock_listdevdir_i3_2xlarge),\
+            unittest.mock.patch('lib.scylla_cloud.run', side_effect=mock_multi_run_i3_2xlarge),\
+            unittest.mock.patch('builtins.open', unittest.mock.MagicMock(side_effect=mock_multi_open_i3_2xlarge)):
+            ins = aws_instance()
+            assert ins._non_root_nvmes() == {'root': ['/dev/xvda1'], 'ephemeral': ['nvme0n1', 'nvme1n1'], 'ebs': []}
+
+    def test___populate_disks_i3_2xlarge(self):
```
Comment (fruch) [inline tests/test_aws_instance.py:450]:
```suggestion
    def test_populate_disks_i3_2xlarge(self):
```

File: tests/test_aws_instance.py [hunk 5]
```diff
@@ -0,0 +1,523 @@
...
+        self.httpretty_aws_metadata(with_ebs=True)
+        with unittest.mock.patch('psutil.disk_partitions', return_value=mock_disk_partitions),\
+            unittest.mock.patch('os.listdir', return_value=mock_listdevdir_i3en_2xlarge_with_ebs),\
+            unittest.mock.patch('lib.scylla_cloud.run', side_effect=mock_multi_run_i3en_2xlarge),\
+            unittest.mock.patch('builtins.open', unittest.mock.MagicMock(side_effect=mock_multi_open_i3en_2xlarge_with_ebs)):
+            ins = aws_instance()
+            assert ins.get_remote_disks() == {'nvme2n1', 'nvme1n1'}
+
+
+    def test__non_root_nvmes_i3_2xlarge(self):
```
Comment (fruch) [inline tests/test_aws_instance.py:441]:
```suggestion
    def test_non_root_nvmes_i3_2xlarge(self):
```

File: common/scylla_image_setup [hunk 6]
```diff
@@ -17,14 +17,13 @@
...
-sys.path.append('/opt/scylladb/scripts')
-from scylla_util import is_redhat_variant
+import distro
 from lib.scylla_cloud import get_cloud_instance, is_gce
 from subprocess import run
 
 if __name__ == '__main__':
     # On Ubuntu, we configure CPU scaling while AMI building time
-    if is_redhat_variant():
+    if distro.id() == 'centos':
```
Comment (fruch) [inline common/scylla_image_setup:26]: this is not as not redhat.... (i.e. now we support images only for centos and ubuntu, but down the road we might have more, i.e. rocky based image ?)

**Review by fruch — APPROVED** [review]
few nipticks, but other than that LGTM

File: tests/test_aws_instance.py (hunk 1 above)
Comment (syuu1228) [reply tests/test_aws_instance.py]: fixed.

File: tests/test_aws_instance.py (hunk 2 above)
Comment (syuu1228) [reply tests/test_aws_instance.py]: fixed.

File: tests/test_aws_instance.py (hunk 3 above)
Comment (syuu1228) [reply tests/test_aws_instance.py]: fixed.

File: tests/test_aws_instance.py (hunk 4 above)
Comment (syuu1228) [reply tests/test_aws_instance.py]: fixed.

File: tests/test_aws_instance.py (hunk 5 above)
Comment (syuu1228) [reply tests/test_aws_instance.py]: fixed.

**Comment by syuu1228** [comment]
Rebased, since it conflicted latest next branch

**Comment by syuu1228** [comment]
@yaronkaikov ping

**Comment by syuu1228** [comment]
> @syuu1228 What is that we testing with test_aws_instance.py?

This tests `class aws_instance` in lib/scylla_cloud.py without using EC2 instance, by mocking EC2 instance metadata URLs and local disk interfaces (NVMe related files on /sys and /dev, etc).
It doesn't emulate entire EC2 instance behavior, but still able to find some kind of bugs.
I actually found two bugs during implementing this, and the patch is included this PR:
 - bug1, syntax error: https://github.com/scylladb/scylla-machine-image/pull/312/commits/69f01ed6c718030b51fc26c2f46e25fb7041401e
 - bug2, got exception when user-data is not available: https://github.com/scylladb/scylla-machine-image/pull/312/commits/6bb409e39914c121627128c9dc52f2567963e11b

**Comment by syuu1228** [comment]
Also implemented gcp_instance test, but I think I need to send it after this PR get merged, since it has dependency to this.
https://github.com/syuu1228/scylla-machine-image/commits/test_gce_instance

**Review by yaronkaikov — APPROVED** [review]

**Pull request merged**